src/app/(proper_react)/redesign/(authenticated)/user/dashboard/Dashboard.test.tsx

```diff
@@ -227,3 +228,26 @@ it("shows returned free user who has resolved all tasks premium upsell and all f
   await user.click(bannerPremiumCta[0]);
   expect(screen.getByRole("dialog")).toBeInTheDocument();
 });
+
+it("shows a returning Premium user who has resolved all tasks a CTA to check out what was fixed", async () => {
+  enablePremium();
+  const user = userEvent.setup();
+  const ComposedDashboard = composeStory(DashboardPremiumUserAllResolved, Meta);
+  render(<ComposedDashboard />);
+
+  // We show a CTA on desktop in the toolbar and in the mobile menu
+  const premiumBadges = screen.queryAllByText("Premium");
+  expect(premiumBadges.length).toBe(2);
+
+  // show banner CTA premium upgrade
+  const bannerPremiumCta = screen.queryAllByRole("button", {
+    name: "See what’s fixed",
+  });
+  expect(bannerPremiumCta.length).toBe(1);
+
+  // click on cta
+  await user.click(bannerPremiumCta[0]);
+
+  const fixedTab = screen.getByRole("tab", { name: "Fixed" });
+  expect(fixedTab).toHaveAttribute("aria-selected", "true");
```

Comment:
This highlighted a bug that I fixed in `TabList.tsx` that resulted in the tab not actually getting selected when switching tabs via the top banner CTA.

Also, the existing tests highlighted two logic errors in `getRelevantGuidedSteps`, so thanks for writing those @codemist and @flozia :)